app/controllers/stats_controller.rb

```diff
@@ -0,0 +1,573 @@
+class StatsController < ApplicationController
+    def donation_totals
+        @result = ActiveRecord::Base.connection.execute(donation_query())
+        return @result
+      end
+
+    def gam_count
+        @GiftCardDetail = GiftCardDetail.where("single_use": true)
+        return @GiftCardDetail.length + 5140
+    end 
+    def index
+        # some other stuff going on...
+        # donation_totals()
+        show2(donation_totals(), sellers_total(), transaction_totals(), gam_count()) 
+    end
+
+    def sellers_total
+        @sellers = Seller.all
+        return @sellers.length()
+    end
+
+    def transaction_totals
+        @Item = Item.all
+        return @Item.length()
+    end
+    
+
+  def donation_query
+    query = <<-SQL 
```

Comment:
this is a huge (kudos for working on it!) query -- any chance we can move this to a `.sql` file somewhere, then load it to a string and execute it as needed? Would be great to have intelligence while editing it and tracking it separately